### DISCUSSION TITLE
Podman - Docker Using LXC Not Recommended by Proxmox Staff

### DISCUSSION THREAD

**Discussion by noackje20** [tteck/Proxmox#1035] [General]
I have been reading a lot about how it is not recommended to use podman or docker in a lxc.  Here is the article from proxmox forum about the issues.  I really love the scripts that are offered here but I think the use of podman in lxc might be a bad idea at this time.

https://forum.proxmox.com/threads/podman-in-lxc-what-do-overlay-not-support-file-handles-and-conflicting-options-userxattr-metacopy-mean.121825/#post-529569

Just wanted to see if we should continue to use the podman lxc home assistant and or podman lxc installer you have or best to just migrate to a VM before any issues arise cause it is working.

Thanks

**Comment by tteck** [comment]
Linux containers have made significant progress and I have never encountered any problems when using Podman or Docker within an LXC environment.

**Reply by tteck** [reply]
As of now, there are only 6 scripts that utilize Docker or Podman.

**Reply by noackje20** [reply]
That's good news that it doesn't seem to be an issue.  I just wanted to point it out that I seen it on the proxmox and verify there wasn't any issues.  Thanks again for the hard work great scripts.

**Reply by tteck** [reply]
It has been a topic of discussion for several years.

**Comment by tazmo** [comment]
I'd like to update this thread please... I also have never had a problem with docker in an lxc. Although it has been "inefficient" since there has not been "support" of overlayfs in zfs (for those that run it on zfs) until recently.

PVE 8.1 now includes openzfs 2.2.2, with fixes for recently discovered, extremely rare/edge-case bug, that supports overlayfs in a more efficient manner (namely with the use of the renameat2(2) function.

**Reply by tteck** [reply]
https://github.com/tteck/Proxmox/commit/a22c3644a69470af17bd60f6f56173b5eb248cb1
Now, Podman doesn't work with ZFS

**Reply by filou59** [reply]
Hello 
Podman seems to work with ZFS by following the following instructions:
https://forum.proxmox.com/threads/podman-on-lxc-with-zfs-backed-volume-and-overlay.138722/
https://medium.com/@rar1871/docker-podman-in-lxc-with-zfs-5196a13bb8be

Without it's Podman instruction throws an error, but once applied it seems to work.

**Comment by rddaz2013** [comment]
First of all, thank you to the makers of Helper-Scripts.com. You have made life easier for me and many others with your scripts.

I switched to Proxmox for my home lab 2 years ago and it works. A problem that is occurring more frequently now is when I started using a ZFS file system for the LXC disk.

The installation scripts do not run error-free (see previous post) - there seems to be a connection with Proxmox/LCX/Docker&Podman but I am not quite sure whether it is due to my Proxmox/ZFS configuration or the installation curtain under Proxmox in general.

My workaround at the moment is first to create the LCX disk locally - then the installation script runs error-free. I would now try to move the disk to the ZFS system.

Is that a possible approach, or can I change something in my Docker/ZFS/Proxmox configuration?

Here are a few more links:

- [podman-on-lxc-with-zfs-backed-volume-and-overlay](https://forum.proxmox.com/threads/podman-on-lxc-with-zfs-backed-volume-and-overlay.138722/)
- [podman-in-lxc-what-do-overlay-not-support-file-handles](https://forum.proxmox.com/threads/podman-in-lxc-what-do-overlay-not-support-file-handles-and-conflicting-options-userxattr-metacopy-mean.121825/#post-529569)
- [probleme-mit-lxc-oder-vm-bei-speicherung-auf-zfs-storage](https://forum.proxmox.com/threads/probleme-mit-lxc-oder-vm-bei-speicherung-auf-zfs-storage.131161/)
